### DISCUSSION TITLE
why do we do rdr2geo and geo2rdr twice in alos2App.py

### DISCUSSION THREAD

**Discussion by domino20** [isce-framework/isce2#738] [Q&A] [answered]
The steps in also2App.py are  'preprocess', 'baseline', 'download_dem', 'prep_slc', 'slc_offset', 'form_int', 'swath_offset', 'swath_mosaic', 'frame_offset', 'frame_mosaic', 'rdr2geo', 'geo2rdr', 'rdrdem_offset', 'rect_rgoffset', 'diff_int', 'look', 'coherence', etc. 
1. I can see that inside the step slc_offset, the rdr2geo and geo2rdr operations have actually been done to get the registration offsets of the secondary image(cpu version used). And then form interferogram by using form_int. What I am confused is that rdr2geo and  
geo2rdr are then used again(we can use gpu version this time), I don`t know why they are used here. 
2. I am also wondering why we need to use rdrdem_offset to calculate the offsets of simulated image based on dem. Does this improve the result? @EJFielding @CunrenLiang

**Comment by CunrenLiang** [comment, marked as answer]
Since L-band data might be significantly affected by ionosphere, cross-correlation is still needed to coregister the SLCs.  rdr2geo is used to find out the water body areas before doing cross-correlation. Steps 'rdr2geo' and 'geo2rdr' are used to compute geometrical offsets and the range offset is used to remove topographic fringes. However, there may be offset between range offset image and radar image, leading to residual topographic phase. This offset is estimated by rdrdem_offset.

 

 

 

 

From: ***@***.***> on behalf of domino20 ***@***.***>
Reply-To: isce-framework/isce2 ***@***.***>
Date: Wednesday, July 5, 2023 at 4:17 PM
To: isce-framework/isce2 ***@***.***>
Cc: CunrenLiang ***@***.***>, Mention ***@***.***>
Subject: [isce-framework/isce2] why do we do rdr2geo and geo2rdr twice in alos2App.py (Discussion #738)

 

The steps in also2App.py are 'preprocess', 'baseline', 'download_dem', 'prep_slc', 'slc_offset', 'form_int', 'swath_offset', 'swath_mosaic', 'frame_offset', 'frame_mosaic', 'rdr2geo', 'geo2rdr', 'rdrdem_offset', 'rect_rgoffset', 'diff_int', 'look', 'coherence', etc.
I can see that inside the step slc_offset, the rdr2geo and geo2rdr operations have actually been done to get the registration offsets of the secondary image(cpu version used). And then form interferogram by using form_int. What I am confused is that rdr2geo and
geo2rdr are then used again(we can use gpu version this time), I don`t know why they are used here.
I am also wondering why we need to use rdrdem_offset to calculate the offsets of simulated image based on dem. Does this improve the result? @EJFielding @CunrenLiang

**Reply by domino20** [reply]
Hi (@CunrenLiang @EJFielding)
I have another question about use  'rdr2geo' and 'geo2rdr' to compute geometrical offsets.  I know the diff interferogram is calculated by substracting this part, but will the calculated range offset be influenced by poor orbit accuracy or strong ionosphere effect?  For a case, I plot the phase calculated with range offset, the fringe looks weird (curved fringe, but the fringe of interferogram is normal). I set the dem to 0 and calculate range offset again and find that the problem is the flat-earth phase, the topographic phase is ok. I don`t know why this happens. And is there any other way to calculate phase by geometrical offsets in isce2
![phase1](https://github.com/isce-framework/isce2/assets/63491281/497e0ccb-ffd6-4564-9788-dc405188d43f)

**Comment by EJFielding** [comment]
For the processing of the NISAR L-band data, we are planning to use a global ionospheric measurement from GNSS data to estimate an extra range offset to be added to the geometric offsets for better coregistration.
